### PR TITLE
first version of framework

### PR DESCRIPTION
**Pull request by yahiko615**

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import os
+
+import pytest
+
+from utilities.config_reader import ReadConfig
+from utilities.driver_factory import create_driver_factory
+from selenium.webdriver.chrome.options import Options
+
+_screenshot_path = r"C:\Users\Admin\Downloads"
```
Comment (bohdandovbsyh) [inline conftest.py:9]: what if I run these tests on Ubuntu? this path will not work

File: conftest.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    driver_options = Options()
+    is_headless = request.node.get_closest_marker("headless")
+    if is_headless:
+        driver_options.add_argument("--headless")
+
+    driver = create_driver_factory(ReadConfig.get_browser_id(), options=driver_options)
+    driver.get(ReadConfig.get_app_battle_pet_page())
+    driver.maximize_window()
+    yield driver
+    driver.quit()
```
Comment (bohdandovbsyh) [inline conftest.py:111]: all these fixtures have a lot of code duplication. Please optimize it

File: page_objects/battle_pet_pack/pet_spell_page.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    __searched_spell_page_locator = (By.XPATH, "//h1[@class='heading-size-1']")
+
+    def move_slider(self):
+        self.move_slider_to(self.__slider_locator, self.__slider_progress_locator, desired_position="50%")
+        return self
+
+    def right_dmg_scaling_is_displayed(self):
+        return self.get_element_style(self.__slider_locator) == "left: 50%;"
+
+    def right_spell_found(self):
```
Comment (bohdandovbsyh) [inline page_objects/battle_pet_pack/pet_spell_page.py:21]: if your function returns True/false its name should starts with is_ 
please find such places all around your project and fix

File: page_objects/dressing_room_page/dressing_room_page.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    __search_result_shoulders_locator = (
+        By.XPATH, "//img[@src='https://wow.zamimg.com/modelviewer/live/webthumbs/item/100/41572.webp']")
+    __new_helm_locator = (By.CSS_SELECTOR, "#dressing-room-paperdoll > div.paperdoll-left > div:nth-child(1) > a")
+    __new_shoulders_locator = (By.CSS_SELECTOR, "#dressing-room-paperdoll > div.paperdoll-left > div:nth-child(3) > a")
+    __shoulders_locator = (By.CSS_SELECTOR, "a[href='https://www.wowhead.com/item=117327/oathsworn-pauldrons']")
+    __replace_locator = (By.XPATH, "//span[text()='Replace...']")
+    __screenshot_button_locator = (By.CSS_SELECTOR, "#dr-screenshot")
+    __save_outfit_button_locator = (By.CSS_SELECTOR, "#dr-save")
+    __toggle_ui_button_locator = (By.CSS_SELECTOR, "#dr-toggle-ui")
+    __downloads_dir = r"C:\Users\Admin\Downloads"
```
Comment (bohdandovbsyh) [inline page_objects/dressing_room_page/dressing_room_page.py:26]: Works only for windows

File: requirements.txt [hunk 5]
```diff
@@ -0,0 +1,3 @@
+pytest
+selenium
+flaky
```
Comment (bohdandovbsyh) [inline requirements.txt:3]: set versions

File: tests/ui_tests/test_battle_pet.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+def test_class_changing(create_black_claw_page):
+    driver = create_black_claw_page
+    pet_ability = PetSpell(driver).move_slider()
+    assert pet_ability.right_dmg_scaling_is_displayed(), "Dmg of ability don't scales wtih the slider value"
+
+
+@pytest.mark.regression
+@pytest.mark.headless
+def test_search_by_spell_name(create_battle_pet_page):
+    driver = create_battle_pet_page
```
Comment (bohdandovbsyh) [inline tests/ui_tests/test_battle_pet.py:18]: why dont return BattlePetAbilities from this Fixture? 
and from all other , why do you return driver?

File: utilities/ui_utilities/base_page.py [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+            else:
+                return False
+        elif page == 'registration':
+            element_text = self.get_text(locator)
+            if text_registration in element_text:
+                return True
+            else:
+                return False
+        else:
+            return False
```
Comment (bohdandovbsyh) [inline utilities/ui_utilities/base_page.py:135]: looks like not a BasePage function

File: conftest.py (hunk 1 above)
Comment (yahiko615) [reply conftest.py:9]: +

File: conftest.py (hunk 2 above)
Comment (yahiko615) [reply conftest.py:111]: +

File: page_objects/battle_pet_pack/pet_spell_page.py (hunk 3 above)
Comment (yahiko615) [reply page_objects/battle_pet_pack/pet_spell_page.py:21]: +

File: page_objects/dressing_room_page/dressing_room_page.py (hunk 4 above)
Comment (yahiko615) [reply page_objects/dressing_room_page/dressing_room_page.py:26]: +

File: requirements.txt (hunk 5 above)
Comment (yahiko615) [reply requirements.txt:3]: +

File: utilities/ui_utilities/base_page.py (hunk 7 above)
Comment (yahiko615) [reply utilities/ui_utilities/base_page.py:135]: +

File: tests/ui_tests/test_battle_pet.py (hunk 6 above)
Comment (yahiko615) [reply tests/ui_tests/test_battle_pet.py:18]: idk, its 2 different pages

Comment (bohdandovbsyh) [reply tests/ui_tests/test_battle_pet.py:17]: I mean that the  main idea is to return Page instances not drivers to every test.

**Review by bohdandovbsyh — APPROVED** [review]

**Pull request merged**